lib/access.js

```diff
@@ -34,152 +20,200 @@ const subcommands = [
   '2fa-not-required',
 ]
 
-const UsageError = (msg) =>
-  Object.assign(new Error(`\nUsage: ${msg}\n\n` + usage), {
-    code: 'EUSAGE',
-  })
-
-const cmd = (args, cb) =>
-  access(args)
-    .then(x => cb(null, x))
-    .catch(err => err.code === 'EUSAGE'
-      ? cb(err.message)
-      : cb(err)
+class Access {
+  constructor (npm) {
+    this.npm = npm
+  }
+
+  // TODO we don't use this, why are we exporting it?
+  get subcommands () {
+    return subcommands
+  }
+
+  get usage () {
```

Comment:
Could we have all of these inherit from a base `Command` class that DRYs up some of the usageUtil()/completion() repetition?  I'm not sure it's paying rent, seems like just another thing for us to forget to do.  The Command constructor could even have some nice safety nets, like:

```js
const _usage = Symbol.for('usage')
class Command {
  constructor (npm) {
    if (!npm)
      throw new TypeError('should pass in npm instance to constructor')
    if (!this[_usage])
      throw new Error('usage not found for command ' + this.constructor.name)
    if (!this[_completion])
      throw new Error('completion not found for command ' + this.constructor.name)
  }
  // override in subclasses with something
  get [_usage] () {
    return null
  }
  get [_completion] () {
    return null
  }
  get usage () {
    return usageUtil(this.name, this[_usage])
  }
  completion (opts) {
    doSomethingWith(this[_completion])
  }
}
```

Reply:
This is a fabulous next step.

Reply:
That test already exists in `test/lib/load-all-commands.js`

Reply:
https://github.com/npm/cli/pull/2821